crates/jstz_api/src/js_log.rs

```diff
@@ -1,13 +1,30 @@
 use boa_engine::{Context, JsNativeError, JsResult};
 use serde::{Deserialize, Serialize};
 use std::cell::Cell;
+use std::str::FromStr;
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
 pub enum LogLevel {
-    ERROR,
-    WARN,
-    INFO,
-    LOG,
+    ERROR = 1,
+    WARN = 2,
+    INFO = 3,
+    LOG = 4,
+}
+
+pub const DEFAULT_LOG_LOG_LEVEL: LogLevel = LogLevel::LOG;
+
+impl FromStr for LogLevel {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_uppercase().as_str() {
+            "ERROR" => Ok(LogLevel::ERROR),
+            "WARN" => Ok(LogLevel::WARN),
+            "INFO" => Ok(LogLevel::INFO),
+            "LOG" => Ok(LogLevel::LOG),
+            _ => Err("no match"),
+        }
+    }
```

Comment:
We can remove all these lines by using a macro :)
```
#[derive(
    Serialize, Deserialize, PartialEq, PartialOrd, Clone, Debug, clap::ValueEnum, Default,
)]
pub enum LogLevel {
    ERROR = 1,
    WARN = 2,
    INFO = 3,
    #[default]
    LOG = 4,
}
```

Reply:
Perhaps we don't have to define the default values here but rather in jstz_cli since log being the default value is cli specific and not api related. Wdyt?

Reply:
One of the drawbacks of this approach is case matching. For instance, my implementation would work with:
```
jstz logs trace tz4MRUzq2DD3dzx2F6sCeeD1kfpK9ejgBsz6 --level WARN
jstz logs trace tz4MRUzq2DD3dzx2F6sCeeD1kfpK9ejgBsz6 --level warn
jstz logs trace tz4MRUzq2DD3dzx2F6sCeeD1kfpK9ejgBsz6 --level wArN
```

Any idea how to reproduce this with direct enum mapping or is it deemed not that important?

Reply:
i think its better to have it like that because capital letters are a bit awkward to type on the cl, and generally not used by convention AFAIU.

Reply:
After spending tremendous amount of time with , I think this is the best way to go even though it doesn't cover all the `#[serde(rename_all = "...")]`,  couldn't make it work to allow strictly uppercase and lowercase flag values as clap was by default case sensitive towards lowercase. Instead, went with `ignore_case = true` in clap arg macro while directly mapping to the LogLevel enum.